### PR TITLE
Kalman covariance without cross-axis influence

### PR DESCRIPTION
**Pull request by nerdCopter**
* Kalman cleanup recommended by Marinus.
* for Non-Helio FC's.
* Needs A/B testing against existing firmware.
* Needs BBL deep analysis.
* Flies well. Zero ill-effect. Potentially better, if not placebo.
* Recommend testing on 6" and 7" quads to feel the difference.

### CODE REVIEW

**Comment by nerdCopter** [comment]
`varStruct.xySumCoVar`
`varStruct.xzSumCoVar`
`varStruct.yzSumCoVar`
_are no longer used either?_

**Comment by gretel** [comment]
@nerdCopter looks like

**Comment by nerdCopter** [comment]
@Quick-Flash, Another option good for testing purposes would be to have an OSD/CLI setting to set a variable `covariance = FULL|SIMPLE` where A/B testing could occur in the field without re-flashing.  if you like that idea, i could put it on my to-do list.

File: src/main/common/kalman.c [hunk 1]
```diff
@@ -139,21 +107,12 @@ FAST_CODE float kalman_process(kalman_t* kalmanState, float input, float target)
...
-    setPoint[Z] = getSetpointRate(Z);
-
-    update_kalman_covariance(input);
-    output[X] = kalman_process(&kalmanFilterStateRate[X], input[X], setPoint[X] );
-    output[Y] = kalman_process(&kalmanFilterStateRate[Y], input[Y], setPoint[Y] );
-    output[Z] = kalman_process(&kalmanFilterStateRate[Z], input[Z], setPoint[Z] );
-
-    DEBUG_SET(DEBUG_KALMAN, 1, input[X]);                               //Gyro input
+    update_kalman_covariance(input, axis);
+    input = kalman_process(&kalmanFilterStateRate[axis], input, getSetpointRate(axis) );
```
Comment (gretel) [inline src/main/common/kalman.c:114]: Codacy Issue found: [Variable 'input' is assigned a value that is never used.](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=5382440)

**Comment by gretel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=5382440&bid=17647836)

File: src/main/common/kalman.c (hunk 1 above)
Comment (gretel) [inline src/main/common/kalman.c:114]: Codacy Issue found: [Assignment of function parameter has no effect outside the function.](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=5382440)

**Comment by gretel** [comment]
please work on the code quality.

**Comment by Quick-Flash** [comment]
> please work on the code quality.

you talking to me?

**Comment by gretel** [comment]
sorry, nah.

**Comment by nerdCopter** [comment]
* flew 4 packs of per-axis mod on foxeerf405.
* no deviation from expectations.
* no known issues.
* flew as well as any other recent flight code.

**Comment by nerdCopter** [comment]
closing.
* recreating non-"onD"  branch.
* included in `testing-kalman` branch.

**Pull request closed without merging**